ntoskrnl/cc/view.c

```diff
@@ -505,6 +504,9 @@ CcRosTrimCache(
             ULONG i;
             PFN_NUMBER Page;
 
+            /* Keep a reference on the VACB */
+            CcRosVacbIncRefCount(current);
```

Comment:
Great idea, except that it only protects `current` and not `current_entry`, which points to the next entry, which is not referenced. (See comment above)